<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MeshEditor</class>
 <widget class="QWidget" name="MeshEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>304</width>
    <height>620</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>300</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>1</number>
   </property>
   <property name="topMargin">
    <number>1</number>
   </property>
   <property name="rightMargin">
    <number>1</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Name:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="meshName"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QListWidget" name="meshPartView">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>200</width>
       <height>0</height>
      </size>
     </property>
     <property name="resizeMode">
      <enum>QListView::Adjust</enum>
     </property>
     <property name="spacing">
      <number>2</number>
     </property>
     <property name="selectionRectVisible">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0,0,0">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <widget class="QPushButton" name="saveMeshButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../EditorResources.qrc">
         <normaloff>:/icons/Icons/save.png</normaloff>:/icons/Icons/save.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="addElementButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../EditorResources.qrc">
         <normaloff>:/icons/Icons/plus-icon.png</normaloff>:/icons/Icons/plus-icon.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeElementButton">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../EditorResources.qrc">
         <normaloff>:/icons/Icons/minus-icon.png</normaloff>:/icons/Icons/minus-icon.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_4">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="DefaultMaterialWidget">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>Default Material</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>6</number>
        </property>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="spacing">
         <number>6</number>
        </property>
        <item>
         <widget class="QPushButton" name="dmSelectColor">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>35</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="dm_colorR">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>255</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="dm_colorG">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>255</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="dm_colorB">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>255</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <property name="verticalSpacing">
         <number>0</number>
        </property>
        <item row="2" column="2">
         <widget class="QDoubleSpinBox" name="dm_specularVal">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QDoubleSpinBox" name="dm_metallicVal">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Roughness:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Metallic:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Specular:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSlider" name="dm_roughnessSlider">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>500</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Emissive:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSlider" name="dm_specularSlider">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>200</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QDoubleSpinBox" name="dm_roughnessVal">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>0.500000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSlider" name="dm_metallicSlider">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSlider" name="dm_emissiveSlider">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QDoubleSpinBox" name="dm_emissiveVal">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="castShadow">
        <property name="text">
         <string>castShadow</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Texture Scale:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="dm_textureScaleSlider">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>500</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="dm_textureScaleVal">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="horizontalSpacing">
         <number>0</number>
        </property>
        <item row="1" column="1">
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="diffuseTex">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="normalTex">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QPushButton" name="materialTex">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="iconSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Diffuse:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Material:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Normal:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="Line" name="line_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="advancedMaterial">
        <property name="title">
         <string>Advanced</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Shader :</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="shaderList"/>
           </item>
           <item>
            <widget class="Line" name="line_5">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="moreAdvancedMat">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>More</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QCheckBox" name="cullFace">
             <property name="text">
              <string>cullFace</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cullBackFace">
             <property name="text">
              <string>cullBackFace</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../EditorResources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>meshPartView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>MeshEditor</receiver>
   <slot>itemSelectionChanged(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dm_colorR</sender>
   <signal>editingFinished()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dm_colorG</sender>
   <signal>editingFinished()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dm_colorB</sender>
   <signal>editingFinished()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addElementButton</sender>
   <signal>clicked()</signal>
   <receiver>MeshEditor</receiver>
   <slot>selectGeometryFromResources()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>260</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>369</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeElementButton</sender>
   <signal>clicked()</signal>
   <receiver>MeshEditor</receiver>
   <slot>removeElement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedMaterial</sender>
   <signal>clicked()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>32</x>
     <y>538</y>
    </hint>
    <hint type="destinationlabel">
     <x>419</x>
     <y>536</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shaderList</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>568</y>
    </hint>
    <hint type="destinationlabel">
     <x>368</x>
     <y>584</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>castShadow</sender>
   <signal>clicked()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>62</x>
     <y>590</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>670</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cullFace</sender>
   <signal>clicked()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>590</y>
    </hint>
    <hint type="destinationlabel">
     <x>153</x>
     <y>681</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cullBackFace</sender>
   <signal>clicked()</signal>
   <receiver>MeshEditor</receiver>
   <slot>updateMaterial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>599</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>657</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>itemSelectionChanged(QModelIndex)</slot>
  <slot>updateMaterial()</slot>
  <slot>selectGeometryFromResources()</slot>
  <slot>removeElement()</slot>
 </slots>
</ui>
